File: server/src/main/java/com/ahmadramadhan/mudahtitip/consignor/dto/GuestConsignorRequest.java
package com.ahmadramadhan.mudahtitip.consignor.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for creating or updating a guest consignor.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class GuestConsignorRequest {

    @NotBlank(message = "Nama penitip wajib diisi")
    @Size(min = 2, max = 100, message = "Nama harus antara 2-100 karakter")
    private String name;

    @NotBlank(message = "Nomor telepon wajib diisi")
    @Size(max = 20, message = "Nomor telepon maksimal 20 karakter")
    private String phone;

    @Size(max = 500, message = "Alamat maksimal 500 karakter")
    private String address;

    @Size(max = 500, message = "Catatan maksimal 500 karakter")
    private String notes;
}

File: client/lib/features/guest_consignor/data/models/guest_consignor_request.dart
import 'package:equatable/equatable.dart';

/// Request DTO for creating/updating a guest consignor.
class GuestConsignorRequest extends Equatable {
  final String name;
  final String phone;
  final String? address;
  final String? notes;

  const GuestConsignorRequest({
    required this.name,
    required this.phone,
    this.address,
    this.notes,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'phone': phone,
      if (address != null && address!.isNotEmpty) 'address': address,
      if (notes != null && notes!.isNotEmpty) 'notes': notes,
    };
  }

  @override
  List<Object?> get props => [name, phone, address, notes];
}
